src/sagemaker/utils.py

```diff
@@ -354,7 +356,12 @@ def repack_model(inference_script, source_directory, model_uri, sagemaker_sessio
             bucket, key = url.netloc, url.path.lstrip('/')
             new_key = key.replace(os.path.basename(key), new_model_name)
 
-            sagemaker_session.boto_session.resource('s3').Object(bucket, new_key).upload_file(new_model_path)
+            try:
+                sagemaker_session.boto_session.resource('s3').Object(bucket, new_key).upload_file(new_model_path)
+            except exceptions.S3UploadFailedError:
+                bucket = sagemaker_session.default_bucket()
```

Comment:
I think we should be always writing to the default bucket.

Feel free to ask for a second opinion.

Reply:
I agree with you. Good point.